contracts/meta-oracles/HistoricalPriceFeedv2.sol

```diff
@@ -0,0 +1,304 @@
+/*
+    Copyright 2019 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.5.7;
+pragma experimental "ABIEncoderV2";
+
+import { Ownable } from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import { ReentrancyGuard } from "openzeppelin-solidity/contracts/utils/ReentrancyGuard.sol";
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { IMedian } from "../external/DappHub/interfaces/IMedian.sol";
+import { LinkedListLibrary } from "./lib/LinkedListLibrary.sol";
+
+
+/**
+ * @title HistoricalPriceFeedv2
+ * @author Set Protocol
+ *
+ * Contract used to store Historical price data from an off-chain oracle
+ */
+contract HistoricalPriceFeedv2 is
```

Comment:
Knit: In our other contracts, I believe we use capital V when signifying version. So `HistoricalPriceFeedV2`